example/bwmf/bwmf_task.go

```diff
@@ -140,55 +140,43 @@ func (t *bwmfTask) initOptUtil() {
 	tParamLen := t.dims.n * t.dims.k
 	dParamLen := t.dims.m * t.dims.k
 
-	t.tParam = op.NewVecParameter(tParamLen)
-	t.dParam = op.NewVecParameter(dParamLen)
+	t.tParam = op.NewVecParameter(int(tParamLen))
+	t.dParam = op.NewVecParameter(int(dParamLen))
 
 	if t.tShard == nil {
 		// XXX: Initialize it random and sparse.
-		t.tShard = &pb.MatrixShard{
-			Row: make([]*pb.MatrixShard_RowData, t.dims.n),
+		t.tShard = &pb.MatrixShard {
+			M: t.dims.n,
+			N: t.dims.k,
+			Val: make([]float32, t.dims.n * t.dims.k),
 		}
-		m1 := int32((10 + t.dims.n) / 10)
-		for r, _ := range t.tShard.Row {
-			t.tShard.Row[r] = &pb.MatrixShard_RowData{At: make(map[int32]float32)}
-			if rand.Int31()%m1 == 0 {
-				for k := 0; k < t.dims.k; k++ {
-					t.tShard.Row[r].At[int32(k)] = rand.Float32()
-				}
-			}
+		for i := uint32(0); i < t.tShard.M * t.tShard.N; i++ {
+			t.tShard.Val[i] = rand.Float32()
 		}
 	}
 	// assign loaded tshard to tparam
-	for r, m := range t.tShard.GetRow() {
-		for c, v := range m.At {
-			t.tParam.Set(r*t.dims.k+int(c), v)
-		}
+	for i := 0; i < int(t.tShard.M * t.tShard.N); i++ {
+		t.tParam.Set(i, t.tShard.Val[i])
 	}
 
 	if t.dShard == nil {
 		// XXX: Initial at 0.0.
-		t.dShard = &pb.MatrixShard{
-			Row: make([]*pb.MatrixShard_RowData, t.dims.m),
+		t.dShard = &pb.MatrixShard {
```

Comment:
Also, if we choose the dense representation of d/tShard, we can do create d/tParam using the same underlying array, i.e `dShard.Val` or `tShard.Val` by adding a new constructor in op/vec_parameter.go.